Packet++/src/NflogLayer.cpp

```diff
@@ -0,0 +1,82 @@
+#define LOG_MODULE PacketLogModuleNflogLayer
+
+#include "NflogLayer.h"
+#include "IPv4Layer.h"
+#include "IPv6Layer.h"
+#include "PayloadLayer.h"
+#include <string.h>
+#include "NullLoopbackLayer.h"
+
+namespace pcpp
+{
+
+uint8_t NflogLayer::getFamily()
+{
+	nflog_header* hdr = getNflogHeader();
+	return hdr->address_family;
+}
+
+std::pair<uint8_t*, int> NflogLayer::getPayload()
+{
+	uint8_t* data = m_Data + sizeof(nflog_header);
+	nflog_tlv* current_tlv = (nflog_tlv*)data;
+
+	uint16_t offset = sizeof(nflog_header);
+
+	while (current_tlv->tlv_type != NFULA_PAYLOAD) {
```

Comment:
Looks like this is a standard TLV structure if I'm not mistaken. What about using parsers in `TLVData.h` for better support/maintainability?

Reply:
@egecetin thanks
I think using parsers in TLVData.h will be a good idea,
I will work on it

Reply:
@egecetin I have a some difficulties about this
in the example pcap file that @seladb added to issue page, format of TLVs are as len/type/data. but in implementation of TLVData.h there is type/len/data

https://github.com/seladb/PcapPlusPlus/blob/3e764a88a8fef38f0007673f2be9dd4f3f9c1597/Packet%2B%2B/header/TLVData.h#L29

So, in assign() method, when filling m_Data, the type/len will become different with the format of NFLOG. and I cannot use getTLVRecord() properly. 

replacing order of type/len in struct TLVRawData will affect other use cases rather than Nflog (because of calling getType())
I appreciate your suggestion.

Reply:
https://wiki.wireshark.org/uploads/__moin_import__/attachments/SampleCaptures/nflog.pcap

Reply:
I think its better to define a struct inside my class as TLVRawData with correct order of variables.

Reply:
> @egecetin I have a some difficulties about this in the example pcap file that @seladb added to issue page, format of TLVs are as len/type/data. but in implementation of TLVData.h there is type/len/data
> 
> https://github.com/seladb/PcapPlusPlus/blob/3e764a88a8fef38f0007673f2be9dd4f3f9c1597/Packet%2B%2B/header/TLVData.h#L29
> 
> So, in assign() method, when filling m_Data, the type/len will become different with the format of NFLOG. and I cannot use getTLVRecord() properly.
> 
> replacing order of type/len in struct TLVRawData will affect other use cases rather than Nflog (because of calling getType())

Looks like I missed that difference you are right. I will review your changes as soon as possible.